packages/mock-block-dock/src/hook-portals/text/use-did-text-prop-change.tsx

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useRef } from "react";
+
+import { MaybePlainOrRichText } from "./shared";
+
+const generateComparableString = (text: MaybePlainOrRichText) =>
+  typeof text === "string" ? text : JSON.stringify(text);
+
+export const useDidTextChange = (text: MaybePlainOrRichText) => {
```

Comment:
I think the result of this is only used the side effect bit that should be an effect, right? It's probably better having this return a function that tells you if its changed – reading from refs during render is a no-no. 

<img width="888" alt="image" src="https://user-images.githubusercontent.com/719814/196926920-6bed53ac-e359-4db3-9a7f-086743de5910.png">

https://beta.reactjs.org/apis/react/useRef#useref

Reply:
How about this? https://github.com/blockprotocol/blockprotocol/pull/656/commits/2896b994ecfbec767fe2f990d9c7e3f74e95005d